plugins/playstation/src/playstation-plugin.vala

```diff
@@ -0,0 +1,38 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStation : Object, Plugin {
+	private const string FINGERPRINT_PREFIX = "playstation";
+	private const string MIME_TYPE = "application/x-cue";
+	private const string MODULE_BASENAME = "libretro-playstation.so";
+	private const bool SUPPORTS_SNAPSHOTTING = true;
+
+	public GameSource get_game_source () throws Error {
+		var query = new MimeTypeTrackerQuery (MIME_TYPE, game_for_uri);
+		var connection = Tracker.Sparql.Connection.@get ();
+		var source = new TrackerGameSource (connection);
+		source.add_query (query);
+
+		return source;
+	}
+
+	private static Game game_for_uri (string uri) throws Error {
+		var cue_file = File.new_for_uri (uri);
+		var cue_sheet_handler = new CueSheetHandler (cue_file);
+		var bin_file = cue_sheet_handler.extract_bin_file ();
+		var validator = new PlayStationHeader (bin_file);
```

Comment:
`header` sounds better, especially when we won't use it just to validate the game but also to get the disc's ID.